pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -158,12 +159,20 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		before = pr.Status.GetCondition(apis.ConditionSucceeded)
 	}
 
+	getPipelineFunc, err := resources.GetPipelineFunc(c.KubeClientSet, c.PipelineClientSet, pr.Spec.PipelineRef, pr.Namespace, pr.Spec.ServiceAccountName)
```

Comment:
`GetPipelineFunc` is called here and then fed to `GetPipelineData` on line 299 and 317. Can we instead make this as part of `GetPipelineData` itself so that retrieving pipeline either by ref or by spec is all in one place. 

The function signature is too long here instead `GetPipelineFunc` can operate directly on `pr` ( I think someone had already commented on this but couldn't find it).

Also, `pr.Spec.PipelineRef` could be `nil` in case of a pipeline specified through `pipelineSpec` and this call is not necessary. 

Moving this function call inside `GetPipelineData` will simplify `pipelinerun.go`.

Reply:
This logic stems from a pr a long time ago where this API was hashed out so I am hesitant to change it here. We could probably address it in a follow on since a rewrite of how resources are injected is probably warranted on the whole.

In this instance though, I think the current code maintains the contract where the `GetPipelineData` is abstracting out how the `Pipeline` instance is fetched to the interface of the `GetPipelineFunc` it calls. In the current implementation, it allows for the `PipelineRef` to be `nil` as well.

Reply:
Agreed that if we want to change that (in some way or another) it should be on its own PR :wink: